daft/context.py

```diff
@@ -24,7 +24,7 @@ class _PyRunnerConfig(_RunnerConfig):
     use_thread_pool: bool | None
 
 
-@dataclasses.dataclass(frozen=True)
+@dataclasses.dataclass()
```

Comment:
Nit: Don't need to call the `dataclass` function if not passing any args.
```suggestion
@dataclasses.dataclass
```